Tests/AsyncHTTPClientTests/HTTPRequestStateMachineTests.swift

```diff
@@ -373,15 +391,21 @@ class HTTPRequestStateMachineTests: XCTestCase {
         let requestHead = HTTPRequestHead(version: .http1_1, method: .GET, uri: "/")
         let metadata = RequestFramingMetadata(connectionClose: false, body: .fixedSize(0))
         XCTAssertEqual(state.startRequest(head: requestHead, metadata: metadata), .sendRequestHead(requestHead, startBody: false))
-        XCTAssertEqual(state.requestCancelled(), .failRequest(HTTPClientError.cancelled, .close))
+        guard case .failRequest(let error, .close) = state.requestCancelled() else {
+            return XCTFail("unexpected action")
+        }
+        XCTAssertEqual(error as? HTTPClientError, .cancelled)
     }
 
     func testRemoteSuddenlyClosesTheConnection() {
         var state = HTTPRequestStateMachine(isChannelWritable: true)
         let requestHead = HTTPRequestHead(version: .http1_1, method: .GET, uri: "/", headers: .init([("content-length", "4")]))
         let metadata = RequestFramingMetadata(connectionClose: false, body: .fixedSize(4))
         XCTAssertEqual(state.startRequest(head: requestHead, metadata: metadata), .sendRequestHead(requestHead, startBody: true))
-        XCTAssertEqual(state.requestCancelled(), .failRequest(HTTPClientError.remoteConnectionClosed, .close))
+        guard case .failRequest(let error, .close) = state.requestCancelled() else {
+            return XCTFail("unexpected action")
+        }
+        XCTAssertEqual(error as? HTTPClientError, .cancelled)
```

Comment:
Previously we checked for `.remoteConnectionClosed` but that was probably wrong and I think we actually  wanted to check for `.cancel`.